crates/krator-derive/src/admission.rs

```diff
@@ -0,0 +1,273 @@
+    pub name: String,
+}
+
+pub(crate) fn run_custom_derive<T>(input: TokenStream) -> TokenStream
+where
+    T: CustomDerive,
+{
+    let input: proc_macro2::TokenStream = input.into();
+    let token_stream = match syn::parse2(input)
+        .and_then(|input| <T as CustomDerive>::parse(input))
+        .and_then(<T as CustomDerive>::emit)
+    {
+        Ok(token_stream) => token_stream,
+        Err(err) => err.to_compile_error(),
+    };
+
+    token_stream.into()
+}
+
+trait ResultExt<T> {
+    fn spanning(self, spanned: impl quote::ToTokens) -> Result<T>;
+}
+
+impl<T, E> ResultExt<T> for std::result::Result<T, E>
+where
+    E: std::fmt::Display,
+{
+    fn spanning(self, spanned: impl quote::ToTokens) -> Result<T> {
+        self.map_err(|err| syn::Error::new_spanned(spanned, err))
+    }
+}
+
+impl CustomDerive for CustomResourceInfos {
+    fn parse(input: DeriveInput) -> Result<Self> {
+        let ident = input.ident;
+
+        // Limit derive to structs
+        let _s = match input.data {
+            Data::Struct(ref s) => s,
+            _ => {
+                return Err(r#"Enums or Unions can not #[derive(AdmisstionWebhook)"#)
+                    .spanning(ident)
+            }
+        };
+
+        // Outputs
+        let mut cri = CustomResourceInfos {
+            name: "".to_string(),
+        };
+
+        let mut name: Option<String> = None;
+
+        // Arg parsing
+        for attr in &input.attrs {
+            if let syn::AttrStyle::Outer = attr.style {
+            } else {
+                continue;
+            }
+            if !attr.path.is_ident("kube") {
+                continue;
+            }
+            let metas = match attr.parse_meta()? {
+                syn::Meta::List(meta) => meta.nested,
+                meta => {
+                    return Err(r#"#[kube] expects a list of metas, like `#[kube(...)]`"#)
+                        .spanning(meta)
+                }
+            };
+
+            for meta in metas {
+                match &meta {
+                    // key-value arguments
+                    syn::NestedMeta::Meta(syn::Meta::NameValue(meta)) => {
+                        if meta.path.is_ident("kind") {
+                            if let syn::Lit::Str(lit) = &meta.lit {
+                                name = Some(lit.value());
+                                break;
+                            } else {
+                                return Err(
+                                    r#"#[kube(kind = "...")] expects a string literal value"#,
+                                )
+                                .spanning(meta);
+                            }
+                        }
+                    } // unknown arg
+                    _ => (),
+                };
+            }
+        }
+        cri.name = name.expect("kube macro must have property name set");
+
+        Ok(cri)
+    }
+
+    fn emit(self) -> Result<proc_macro2::TokenStream> {
```

Comment:
There is a lot of stuff here that I think we may want to be able to configure via attributes.

Reply:
attributes to the feature? good idea! Can you switch on/off dependencies with this? The only real gain would probably less dependencies if, for example, you are not interested in the automatic generation of the certificate & private key.

Reply:
I will have to look into how to parse those attributes

Reply:
this is implemented in commit 193b6ac